common.go

```diff
@@ -0,0 +1,110 @@
+package cose
+
+import (
+	"errors"
+	"fmt"
+)
+
+// IntOrStr is a value that can be
+type IntOrStr struct {
+	intVal *int
```

Comment:
Shouldn't we use one variable of interface type and set it from only allowed integer and string and also return accordingly ?

Reply:
It would work, but as we know that there are only two possibilities here, we can avoid boxing objects by being a little clever. See https://github.com/veraison/go-cose/pull/146/files#r1221706908.

Reply:
That would be an alternative way of doing it. The problem with that is that you'd have to time resolution inside pretty much every method; this way, it's only done on creation. It's a trade off :shrug: